Description: Fix build failure on GNU/Hurd
 The scenario is the following: in avahi-core, three different network
 interface backends,
 - iface-linux.c (compiled if linux/netlink.h is found)
 - iface-pfroute.c (compiled if net/route.h is found)
 - iface-none.c (compiled in the remaining cases)
 on Hurd, it finds net/route.h so it tries to compile the pfroute backend; the
 problem is that the backend uses net/if_dl.h and the rt_* network stuff,
 unavailable on the Hurd.
 The proposed patch extends the configure check for net/route.h by adding
 net/if_dl.h among the includes needed for the check, so it will pass if both
 net/route.h and net/if_dl.h are available.
Author: Pino Toscano <pino@kde.org>
Debian-Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=558975
Bug: http://avahi.org/ticket/298
Index: b/configure.ac
===================================================================
--- a/configure.ac	2010-06-25 11:30:46.990932953 +0200
+++ b/configure.ac	2010-06-25 11:31:06.274935613 +0200
@@ -169,6 +169,7 @@
 , [], [
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <net/if_dl.h>
 ])
 
 AM_CONDITIONAL(HAVE_PF_ROUTE, [ test x"$HAVE_PF_ROUTE" = xyes ])
